File: python/TiketFilm.py
from Tiket import Tiket


class TiketFilm(Tiket):
    def __init__(
        self,
        idTiket,
        nomorKursi,
        harga,
        statusTiket,
        judulFilm,
        genre,
        durasi,
        ratingUsia
    ):
        super().__init__(
            idTiket,
            nomorKursi,
            harga,
            statusTiket
        )

        self.__judulFilm = str(judulFilm)
        self.__genre = str(genre)
        self.__durasi = int(durasi)
        self.__ratingUsia = int(ratingUsia)

    def getJudulFilm(self):
        return self.__judulFilm

    def getGenre(self):
        return self.__genre

    def getDurasi(self):
        return self.__durasi

    def getRatingUsia(self):
        return self.__ratingUsia

    def setJudulFilm(self, judulFilm):
        self.__judulFilm = str(judulFilm)

    def setGenre(self, genre):
        self.__genre = str(genre)

    def setDurasi(self, durasi):
        self.__durasi = int(durasi)

    def setRatingUsia(self, ratingUsia):
        self.__ratingUsia = int(ratingUsia)

File: java/TiketFilm.java
public class TiketFilm extends Tiket {
    private String judulFilm;
    private String genre;
    private int durasi;
    private int ratingUsia;

    public TiketFilm(
        String idTiket,
        String nomorKursi,
        int harga,
        String statusTiket,
        String judulFilm,
        String genre,
        int durasi,
        int ratingUsia
    ) {
        super(idTiket, nomorKursi, harga, statusTiket);

        this.judulFilm = judulFilm;
        this.genre = genre;
        this.durasi = durasi;
        this.ratingUsia = ratingUsia;
    }

    public String getJudulFilm() {
        return judulFilm;
    }

    public String getGenre() {
        return genre;
    }

    public int getDurasi() {
        return durasi;
    }

    public int getRatingUsia() {
        return ratingUsia;
    }

    public void setJudulFilm(String judulFilm) {
        this.judulFilm = judulFilm;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public void setDurasi(int durasi) {
        this.durasi = durasi;
    }

    public void setRatingUsia(int ratingUsia) {
        this.ratingUsia = ratingUsia;
    }
}